### ISSUE TITLE
Nix install on x86_64 Fedora 35 fails

### ISSUE THREAD

**Issue by ConnorBaker** [NixOS/nix#6080]
**Describe the bug**

I'm unable to install Nix on my x86_64 machine running a brand-new install of Fedora 35. I've wiped the disk and tried to re-install multiple times (including on the Fedora 36 beta, as well as Rawhide), but nothing has worked yet.

Following the directions on nixos.org, I get the following log: https://gist.github.com/ConnorBaker/70457b999a805764d6a71c2eaf151faf

**Steps To Reproduce**

1. Install Fedora and open a terminal
    - ~~EDITED: Run `sudo dnf update` or some other command which prompts for sudo password~~ Unrelated
2. `sh <(curl -L https://nixos.org/nix/install) --daemon`
3. Answer yes to the prompts, allowing the installer to use `sudo`

**Expected behavior**

Nix should be successfully installed.

**Additional context**

```console
$ lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         39 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  8
  On-line CPU(s) list:   0-7
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
    CPU family:          6
    Model:               60
    Thread(s) per core:  2
    Core(s) per socket:  4
    Socket(s):           1
    Stepping:            3
    CPU max MHz:         4000.0000
    CPU min MHz:         800.0000
    BogoMIPS:            7632.80
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonst
                         op_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm cpuid_fault
                          epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid xsaveopt dtherm ida arat pln pts md_clear flush_l1d
Virtualization features: 
  Virtualization:        VT-x
Caches (sum of all):     
  L1d:                   128 KiB (4 instances)
  L1i:                   128 KiB (4 instances)
  L2:                    1 MiB (4 instances)
  L3:                    8 MiB (1 instance)
NUMA:                    
  NUMA node(s):          1
  NUMA node0 CPU(s):     0-7
Vulnerabilities:         
  Itlb multihit:         KVM: Mitigation: VMX disabled
  L1tf:                  Mitigation; PTE Inversion; VMX conditional cache flushes, SMT vulnerable
  Mds:                   Mitigation; Clear CPU buffers; SMT vulnerable
  Meltdown:              Mitigation; PTI
  Spec store bypass:     Mitigation; Speculative Store Bypass disabled via prctl
  Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:            Mitigation; Full generic retpoline, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling
  Srbds:                 Mitigation; Microcode
  Tsx async abort:       Not affected
$ lsmem
RANGE                                  SIZE  STATE REMOVABLE  BLOCK
0x0000000000000000-0x00000000dfffffff  3.5G online       yes   0-27
0x0000000100000000-0x000000081fffffff 28.5G online       yes 32-259

Memory block size:       128M
Total online memory:      32G
Total offline memory:      0B
$ uname -a
Linux fedora 5.16.7-200.fc35.x86_64 #1 SMP PREEMPT Sun Feb 6 19:53:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by ConnorBaker** [comment]
This morning, I re-installed Fedora and ran the same install command, but this time it prompted me for my sudo password. The log I posted in the gist shows that it asked if it could use sudo, *but never asked for the password to elevate permissions!*

I was able to reproduce the issue I had run into previously by running `sudo dnf update`, which prompts me for my sudo password, immediately before attempting to install nix. I get the same log as in the gist: asks if it can use sudo but doesn’t ask for the password.

Is this related to Fedora only asking for sudo for the first command to use it within some timeframe? For example, subsequent `sudo dnf update` or other commands which are run with sudo don’t ask for the password for a while after providing it.
